client/notifications/notification-reducer.ts

```diff
@@ -35,26 +33,52 @@ function toNotificationRecord(notification: Readonly<Notification>): Notificatio
 }
 
 export class NotificationState extends Record({
-  list: List<NotificationRecord>(),
-  ids: Set<string>(),
+  map: Map<string, NotificationRecord>(),
+  ids: List<string>(),
 }) {}
 
 export default keyedReducer(new NotificationState(), {
+  ['@notifications/serverInit'](state, { payload: { notifications } }) {
+    return state
+      .update('map', m => m.merge(notifications.map(n => [n.id, toNotificationRecord(n)])))
```

Comment:
I thought the `merge` method will already do that though? At least from what I can tell by reading the documentation, it should just update the existing keys to the new value. Unless you're talking about something else that I'm missing here.

Reply:
This comment is on the line above the relevant one, sorry. Below this you're unshifting all of the IDs into a list, regardless of whether they are new to the map. So yes, the map will only have one version of each notification, but you might reference that notification 2+ times (and thus it will be displayed 2+ times).

Reply:
You marked this as resolved but the problem still exists?

Reply:
Ah nevermind, I see you changed it to an OrderedSet, that's fine.